covid_graphs/covid_graphs/scripts/server.py

```diff
@@ -56,20 +36,27 @@ def home():
     def covid19_redirect():
         return redirect(url_for("covid19_predictions", event="apr12", graph_type="normal"))
 
-    i = inotify.adapters.InotifyTree(str(data_dir))
-
-    p = Process(target=server.run, kwargs=dict(host="0.0.0.0", port=8081))
+    i = adapters.InotifyTree(str(data_dir),
+                             mask=(constants.IN_MODIFY | constants.IN_DELETE
+                                   | constants.IN_CREATE))
+    p = Process(target=_run_flask_server, kwargs=dict(server=server, data_dir=data_dir))
     p.start()
     while (True):
         events = list(i.event_gen(yield_nones=False, timeout_s=1))
         if len(events) != 0:
             print("Data changed. Restarting server...")
             p.terminate()
             p.join()
-            p = Process(target=server.run, kwargs=dict(host="0.0.0.0", port=8081))
+            p = Process(target=_run_flask_server, kwargs=dict(server=server, data_dir=data_dir))
             p.start()
             print("Server restarted.")
-        sleep(60)
+        sleep(10)
```

Comment:
So we watch for changes in the server code? I thought we'd have an external watch script that would kill and restart the python server process.

Reply:
Initially when we used YAML and it took 2 minutes to reload the server, we had plans to reload partial things like individual graphs for countries that have changed.

I think the external script just adds in more complications and Python can do the same job.